pandas/tests/test_categorical.py

```diff
@@ -152,6 +152,12 @@ def test_is_equal_dtype(self):
             CategoricalIndex(c1, categories=list('cab'))))
         assert not c1.is_dtype_equal(CategoricalIndex(c1, ordered=True))
 
+        # GH 16659
+        s1 = pd.Series(c1)
+        assert c1.is_dtype_equal(s1)
+        assert not c2.is_dtype_equal(s1)
```

Comment:
can u add some tests for things that are not is_dryoe_equal (i don't remember if these are sufficiently covered)

e.g. pass in scalers, ndarray, Dataframe

Reply:
can you update for this